app/src/app/[lng]/[inventory]/InventoryCalculationTab.tsx

```diff
@@ -0,0 +1,165 @@
+                  <SegmentedProgress
+                    values={[thirdPartyProgress, uploadedProgress]}
+                    colors={["interactive.connected", "interactive.tertiary"]}
+                  />
+                  <Heading
+                    fontWeight="semibold"
+                    fontSize="body.md"
+                    className="whitespace-nowrap"
+                  >
+                    {formatPercent(totalProgress)}%{" "}
+                    <Trans t={t}>completed</Trans>
+                  </Heading>
+                </Box>
+                <Box className="flex gap-4 mt-2">
+                  <Tag>
+                    <TagLeftIcon
+                      as={CircleIcon}
+                      boxSize={6}
+                      color="interactive.connected"
+                    />
+                    <TagLabel>
+                      {formatPercent(thirdPartyProgress)}%{" "}
+                      <Trans t={t}>connect-third-party-data</Trans>
+                    </TagLabel>
+                  </Tag>
+                  <Tag>
+                    <TagLeftIcon
+                      as={CircleIcon}
+                      boxSize={6}
+                      color="interactive.tertiary"
+                    />
+                    <TagLabel>
+                      {formatPercent(uploadedProgress)}%{" "}
+                      <Trans t={t}>uploaded-data</Trans>
+                    </TagLabel>
+                  </Tag>
+                </Box>
+                <Box className=" flex flex-col gap-[24px] py-[24px]">
+                  <Text
+                    fontFamily="heading"
+                    fontSize="title.md"
+                    fontWeight="semibold"
+                    lineHeight="24"
+                    my={4}
+                    textColor={"blue"}
+                  >
+                    <Trans t={t}>sector-data</Trans>
+                  </Text>
+                  <Text
+                    fontFamily="heading"
+                    fontSize="title.md"
+                    fontWeight="semibold"
+                    lineHeight="24"
+                  >
+                    <Trans t={t}>sector-emissions</Trans>
+                  </Text>
+                  <Text>
+                    <Trans t={t}>view-progress-in-each-sector</Trans>
+                  </Text>
+                  {isUserInfoLoading || isInventoryProgressLoading ? (
+                    <Center>
+                      <Spinner size="lg" />
+                    </Center>
+                  ) : (
+                    inventoryProgress?.sectorProgress
+                      .slice()
+                      .filter((sectorProgress) => {
+                        return ["I", "II", "III"].includes(
+                          sectorProgress.sector.referenceNumber || "",
+                        );
+                      })
+                      .sort(sortSectors)
+                      .map((sectorProgress, i) => (
```

Comment:
It would be nice to extract this logic to the top of the file and reference the result here in the returned markup.